### PR TITLE
[palette_generator] Add web support to unit tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates the unit tests to run in web mode as well as VM mode, since the package supports web. To remove the file dependency, this adds base64-encoded versions of the existing test images. The originals are left as source-of-truth, in case we want to handle this differently in the future, or to visually inspect the images.

Fixes https://github.com/flutter/flutter/issues/129839

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by stuartmorgan** [comment]
Overrides: this is flagging the removal of `path` from the dependencies, but that's uninteresting. (It should actually have been a dev dependency in the first place.)

File: packages/palette_generator/test/encoded_images.dart [hunk 1]
```diff
@@ -0,0 +1,1351 @@
...
+/// Data version of assets/tall_blue.png.
+final Uint8List kImageDataTallBlue = base64.decode(
+    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAPoCAIAAACQ1AMJAAAACXBIWXMAAAsTAAALEwEAmpwYAA'
+    'AAH0lEQVRIx+3DQQkAAAgAscP+nTWGnw1W7VSqqqrq3wOa1wjO6jYVpgAAAABJRU5ErkJggg='
+    '=');
+
+/// Data version of assets/wide_red.png.
+final Uint8List kImageDataWideRed = base64.decode(
+    'iVBORw0KGgoAAAANSUhEUgAAA+gAAAABCAIAAADCYhNkAAAACXBIWXMAAAsTAAALEwEAmpwYAA'
+    'AAG0lEQVRIx+3BMQEAAAwCoNk/9KzhAeQPAABYV8RfAQE8QBqiAAAAAElFTkSuQmCC');
```
Comment (ditman) [inline packages/palette_generator/test/encoded_images.dart:29]: All the images look good to me (visually), verified with this:

https://jsfiddle.net/ditman/fb8qLmku/

**Review by ditman — APPROVED** [review]
LGTM, thanks for getting this tested on the web!

**Comment by ditman** [comment]
Too late! (x_x)

File: packages/palette_generator/test/encoded_images.dart [hunk 2]
```diff
@@ -0,0 +1,1351 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+// This file contains base64-ecoded versions of the images in assets/, so that
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/palette_generator/test/encoded_images.dart:8]: > We could make a little utility tool to do the encoding and formatting though if it comes up again.

I thought about this as I was writing the reverse one in JSFiddle, it'd be super quick to write in JS, and even give you the formatted output :P

**Pull request merged**